### ISSUE TITLE
Cleaning a queue with a corrupted job will lead to an error

### ISSUE THREAD

**Issue by jlukanta** [OptimalBits/bull#273]
I'm getting "SyntaxError: Unexpected token u at Object.parse" error when calling queue.clean() if a job is that to be removed is corrupted.

I think queue.clean() should still proceed even if a job is corrupted. Otherwise, the job will be stuck forever in the queue.

> 2016-03-23T00:21:52.154Z - error: [SyntaxError: Unexpected token u] { stack: 'SyntaxError: Unexpected token 
>     at Object.parse (native)
>     at Function.Job.fromData (/Users/sanitized/node_modules/bull/lib/job.js:546:40)
>     at /Users/sanitized/node_modules/bull/lib/job.js:50:18
> 
> From previous event:
>     at Job.fromId (/Users/sanitized/node_modules/bull/lib/job.js:48:56)
>     at Array.map (native)
>     at /Users/sanitized/node_modules/bull/lib/queue.js:776:29
>     at RedisClient.return_reply (/Users/sanitized/node_modules/redis/index.js:664:25)
>     at HiredisReplyParser.reply_parser.send_reply (/Users/sanitized/node_modules/redis/index.js:332:14)
>     at HiredisReplyParser.execute (/Users/sanitized/node_modules/redis/lib/parsers/hiredis.js:30:18)
>     at Socket.<anonymous> (/Users/sanitized/node_modules/redis/index.js:131:27)
>     at emitOne (events.js:77:13)
>     at Socket.emit (events.js:169:7)
>     at readableAddChunk (_stream_readable.js:146:16)
>     at Socket.Readable.push (_stream_readable.js:110:10)
>     at TCP.onread (net.js:523:20)
> 
> From previous event:
>     at Queue.getJobs (/Users/sanitized/node_modules/bull/lib/queue.js:775:15)
>     at Queue.getCompleted (/Users/sanitized/node_modules/bull/lib/queue.js:748:15)
>     at /Users/sanitized/node_modules/bull/lib/queue.js:822:18
> From previous event:
>     at Queue.clean (/Users/sanitized/node_modules/bull/lib/queue.js:801:10)
>     at Queue.<anonymous> (/Users/sanitized/thumbnailWorker.js:694:11)
>     at emitNone (events.js:67:13)
>     at Queue.emit (events.js:166:7)
>     at /Users/sanitized/node_modules/bull/lib/queue.js:130:11
>     at RedisClient.return_reply (/Users/sanitized/node_modules/redis/index.js:689:33)
>     at HiredisReplyParser.reply_parser.send_reply (/Users/sanitized/node_modules/redis/index.js:332:14)
>     at HiredisReplyParser.execute (/Users/sanitized/node_modules/redis/lib/parsers/hiredis.js:30:18)
>     at Socket.<anonymous> (/Users/sanitized/node_modules/redis/index.js:131:27)
>     at emitOne (events.js:77:13)
>     at Socket.emit (events.js:169:7)
>     at readableAddChunk (_stream_readable.js:146:16)
>     at Socket.Readable.push (_stream_readable.js:110:10)
>     at TCP.onread (net.js:523:20)
> 
> ...

**Comment by jlukanta** [comment]
The error is thrown while parsing data.data object that is returned from redis:

```
Job.fromData = function(queue, jobId, data){
  var job = new Job(queue, jobId, JSON.parse(data.data), JSON.parse(data.opts));
  ...
```

Not sure why the job got corrupted but this is the data from redis:
```
127.0.0.1:6379> hgetall bull:sanitized:1
1) "attemptsMade"
2) "1"
3) "stacktrace"
4) "[]"
5) "returnvalue"
6) "undefined"
```

**Comment by manast** [comment]
the reason the data got corrupted is a bug that has been fixed in master, but still we need to be able to delete corrupted jobs.

**Comment by jlukanta** [comment]
@manast Thanks for letting me know. I have made a pull request that fixes this issue.

**Comment by jlukanta** [comment]
OK, I will try fixing this issue.

**Comment by jlukanta** [comment]
Fixed. I'm not sure why Travis failed with this error though:
>   1) Priority queue process stalled jobs when starting a queue:
>      Error: timeout of 5000ms exceeded. Ensure the done() callback is being called in this test.
>

**Comment by jlukanta** [comment]
Note: I have seen this error in other merged pull requests. My fix sanitizes invalid data and opt objects from redis in case that a corrupted data got stored. It shouldn't cause this error.

**Comment by chuym** [comment]
Thanks for the patch @jlukanta

**Comment by manast** [comment]
Is this issue resolved in 1.1.2 or master?

**Comment by manast** [comment]
This issue seems to have been fixed some time ago. Reopen if you can still reproduce it.
